pixels-turbo/pixels-worker-vhive/src/test/java/TestStreamWorker.java

```diff
@@ -0,0 +1,632 @@
+        // 6 partition workers (5 for orders table + 1 for customer table) fully connected to 3 partitioned join workers
+        System.out.println("NOTE: 跑之前删一下minio上面的中间结果");
+
+        WorkerContext contextScan = new WorkerContext(LogManager.getLogger(BasePartitionWorker.class), new WorkerMetrics(), "0");
+        PartitionWorker partitionWorker1 = new PartitionWorker(contextScan);
+        PartitionWorker partitionWorker2 = new PartitionWorker(contextScan);
+        PartitionWorker partitionWorker3 = new PartitionWorker(contextScan);
+        PartitionWorker partitionWorker4 = new PartitionWorker(contextScan);
+        PartitionWorker partitionWorker5 = new PartitionWorker(contextScan);
+        PartitionWorker partitionWorker6 = new PartitionWorker(contextScan);
+
+        WorkerContext contextParJoin = new WorkerContext(LogManager.getLogger(BasePartitionedJoinWorker.class), new WorkerMetrics(), "0");
+        PartitionedJoinWorker parJoinWorker1 = new PartitionedJoinWorker(contextParJoin);
+        PartitionedJoinWorker parJoinWorker2 = new PartitionedJoinWorker(contextParJoin);
+        PartitionedJoinWorker parJoinWorker3 = new PartitionedJoinWorker(contextParJoin);
+
+        ExecutorService executor = Executors.newFixedThreadPool(9);
+        Future<JoinOutput>[] parJoinFutures = new Future[] {
+                executor.submit(() -> parJoinWorker1.process(test2GenParJoinInputNonPipelined(minioEndpoint, minioEndpoint, 0))),
+                executor.submit(() -> parJoinWorker2.process(test2GenParJoinInputNonPipelined(minioEndpoint, minioEndpoint, 1))),
+                executor.submit(() -> parJoinWorker3.process(test2GenParJoinInputNonPipelined(minioEndpoint, minioEndpoint, 2)))
+        };
+        Future<PartitionOutput>[] scanFutures = new Future[] {
+                executor.submit(() -> partitionWorker1.process(test2GenParInputNonPipelined(minioEndpoint, minioEndpoint, 0))),
+                executor.submit(() -> partitionWorker2.process(test2GenParInputNonPipelined(minioEndpoint, minioEndpoint, 1))),
+                executor.submit(() -> partitionWorker3.process(test2GenParInputNonPipelined(minioEndpoint, minioEndpoint, 2))),
+                executor.submit(() -> partitionWorker4.process(test2GenParInputNonPipelined(minioEndpoint, minioEndpoint, 3))),
+                executor.submit(() -> partitionWorker5.process(test2GenParInputNonPipelined(minioEndpoint, minioEndpoint, 4))),
+                executor.submit(() -> partitionWorker6.process(test2GenParInputNonPipelined(minioEndpoint, minioEndpoint, 5)))
+        };
+
+        for (Future<JoinOutput> future: parJoinFutures) {
+            JoinOutput parJoinOutput = future.get();
+            System.out.println(JSON.toJSONString(parJoinOutput));
+        }
+
+        executor.shutdown();
+        System.out.println("NOTE: 跑之前删过minio上面的中间结果了吗？");
+    }
+
+    /**
+     * XXX: In the current test, there is only 1 row group from each endpoint. Might need to modify in the future.
```

Comment:
What is XXX?

Reply:
XXX in a code comment means something yet to be discussed or improved. A weak version of TODO. https://stackoverflow.com/questions/1452934/what-is-the-meaning-of-xxx-in-code-comments